microservices/matchmaking-service/src/scrim/scrim.controller.ts

```diff
@@ -92,7 +92,10 @@ export class ScrimController {
     @MessagePattern(MatchmakingEndpoint.SetScrimLocked)
     async setScrimLocked(@Payload() payload: unknown): Promise<boolean> {
         const data = MatchmakingSchemas.SetScrimLocked.input.parse(payload);
-        await this.scrimService.setScrimLocked(data.scrimId, data.locked);
+
+        if (data.locked) await this.scrimService.setScrimLocked(data.scrimId, data.locked, data.reason);
+        else await this.scrimService.setScrimLocked(data.scrimId, data.locked);
+
```

Comment:
This change may not be needed; if data.reason is undefined we can safely pass it.

Reply:
It errored that it didn’t exists because of the union and if being false, TS didn’t like it.